public/app/plugins/panel/bargauge/BarGaugePanel.test.tsx

```diff
@@ -1,106 +1,134 @@
-import { mount, ReactWrapper } from 'enzyme';
+import { render, screen } from '@testing-library/react';
+import { uniqueId } from 'lodash';
 import React from 'react';
 
-import {
-  dateMath,
-  dateTime,
-  FieldConfigSource,
-  LoadingState,
-  PanelData,
-  PanelProps,
-  TimeRange,
-  toDataFrame,
-  VizOrientation,
-} from '@grafana/data';
+import { dateMath, dateTime, EventBus, LoadingState, TimeRange, toDataFrame, VizOrientation } from '@grafana/data';
 import { selectors } from '@grafana/e2e-selectors';
 import { BarGaugeDisplayMode } from '@grafana/ui';
 
-import { BarGaugePanel } from './BarGaugePanel';
-import { PanelOptions } from './models.gen';
+import { BarGaugePanel, BarGaugePanelProps } from './BarGaugePanel';
 
 const valueSelector = selectors.components.Panels.Visualization.BarGauge.valueV2;
 
 describe('BarGaugePanel', () => {
-  describe('when empty result is rendered', () => {
-    const wrapper = createBarGaugePanelWithData({
-      series: [],
-      timeRange: createTimeRange(),
-      state: LoadingState.Done,
-    });
+  describe('when there is no data', () => {
+    it('show a "No Data" message', () => {
+      const panelData = buildPanelData();
+
+      render(<BarGaugePanel {...panelData} />);
 
-    it('should render with title "No data"', () => {
-      const displayValue = wrapper.find(`div[data-testid="${valueSelector}"]`).text();
-      expect(displayValue).toBe('No data');
+      expect(screen.getByText(/no data/i)).toHaveTextContent('No data');
     });
   });
 
   describe('when there is data', () => {
-    const wrapper = createBarGaugePanelWithData({
-      series: [
-        toDataFrame({
-          target: 'test',
-          datapoints: [
-            [100, 1000],
-            [100, 200],
+    it('shows the panel', () => {
+      const firstBarPanel = 'firstBarPanel';
+      const secondBarPanel = 'secondBarPanel';
+      const panelData = buildPanelData({
+        data: {
+          series: [
+            toDataFrame({
+              target: firstBarPanel,
+              datapoints: [
+                [100, 1000],
+                [100, 200],
+              ],
+            }),
           ],
-        }),
-      ],
-      timeRange: createTimeRange(),
-      state: LoadingState.Done,
-    });
+          timeRange: createTimeRange(),
+          state: LoadingState.Done,
+        },
+      });
+
+      const { rerender } = render(<BarGaugePanel {...panelData} />);
+      expect(screen.queryByText(/100/)).toBeInTheDocument();
+      expect(screen.queryByText(/firstbarpanel/i)).not.toBeInTheDocument();
+      expect(screen.queryAllByTestId(valueSelector).length).toEqual(1);
```

Comment:
```suggestion
      expect(screen.getAllByTestId(valueSelector)).toHaveLength(1);
```
Although, if we only expect one element, `expect(screen.getByTestId(valueSelector)).toBeInTheDocument();` could be better here.